lua/neotest-golang/lib/cmd.lua

```
@@ -18,8 +18,9 @@ function M.golist_data(cwd)
   }
   local go_list_command_concat = table.concat(go_list_command, " ")
   logger.debug("Running Go list: " .. go_list_command_concat .. " in " .. cwd)
-  local output =
-    vim.fn.system("cd " .. cwd .. " && " .. go_list_command_concat, " ")
+  local output = vim
+    .system(go_list_command, { cwd = cwd, text = true })
+    :wait().stdout or ""
```

If the command fails, we might as well stop here instead of returning `""`.

For now, let's just warn about the output being `""`, something like this would do:

```lua
if output == "" then
  logger.error({"Execution of 'go list' failed, output:", output})
end
```

---

I just added this in 😄

---

Thanks for the review! Adding log here is indeed useful, we could maybe also show the contents of stderr or exit code here instead of just `output` since it's always an empty string anyway. But I'll leave that to you, it's already good enough for me as it is now 😁

---

That makes a whole lot of sense. I didn't look into vim.system in that regard. Feel free to make a PR which does just this if you like 👍